### PR TITLE
Clean up dangling MCM secret during shoot migration.

### PR DESCRIPTION
**Pull request by Gerrit91**

### CODE REVIEW

**Review by mwennrich — APPROVED** [review]

File: pkg/controller/infrastructure/actuator_migrate.go [hunk 1]
```diff
@@ -2,11 +2,52 @@ package infrastructure
...
+	}
+
+	for _, secret := range secrets.Items {
+		secret := secret.DeepCopy()
+
+		if !strings.HasPrefix(secret.Name, "shoot--") {
+			continue
+		}
+
+		if controllerutil.ContainsFinalizer(secret, deleteFinalizerName) {
```
Comment (majst01) [inline pkg/controller/infrastructure/actuator_migrate.go:40]: Why was the check for deletionTimestamp exists removed ?

Comment (Gerrit91) [reply pkg/controller/infrastructure/actuator_migrate.go:40]: The deletion timestamp is not set at this point in time. It's probably set during namespace deletion.

This should not be any problem though as the MCM does not seem to manage the secret anyway.

**Comment by majst01** [comment]
maybe we should open a issue in g/mcm that this issue will be fixed upstream eventually

**Comment by Gerrit91** [comment]
Probably yes, do you want to file an issue? Maybe it's even fixed already. No matter what, it should not block this PR.

**Comment by majst01** [comment]
> Probably yes, do you want to file an issue? Maybe it's even fixed already. No matter what, it should not block this PR.

this PR is ready to get merged, i already checked in MCM but got confused even more

**Review by majst01 — APPROVED** [review]

**Pull request merged**